packages/pwa-kit-dev/src/utils/script-utils.ts

```diff
@@ -279,6 +279,58 @@ export class CloudAPIClient {
         const data = await res.json()
         return data['token']
     }
+
+    /** Polls MRT for deployment status every 30 seconds. */
+    async waitForDeploy(project: string, environment: string): Promise<void> {
+        return new Promise((resolve, reject) => {
+            /** Milliseconds to wait between checks. */
+            const delay = 30e3
+            /** Check the deployment status to see whether it has finished. */
+            const check = async (): Promise<void> => {
+                const url = new URL(
+                    `/api/projects/${project}/target/${environment}`,
+                    this.opts.origin
+                )
+                const res = await this.opts.fetch(url, {headers: await this.getHeaders()})
+
+                if (!res.ok) {
+                    const text = await res.text()
+                    let json
+                    try {
+                        if (text) json = JSON.parse(text)
+                    } catch (_) {} // eslint-disable-line no-empty
+                    const message = json?.detail ?? text
+                    // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+                    const detail = message ? `: ${message}` : ''
+                    throw new Error(`${res.status} ${res.statusText}${detail}`)
+                }
+
+                const data = await res.json()
+                if (typeof data.state !== 'string') {
+                    return reject(
+                        new Error('An unknown state occurred when polling the deployment.')
+                    )
+                }
+                switch (data.state) {
+                    case 'CREATE_IN_PROGRESS':
+                    case 'PUBLISH_IN_PROGRESS':
+                        // In progress - check again after the next delay
+                        // `check` is async, so we need to use .catch to properly handle errors
+                        setTimeout(() => void check().catch(reject), delay)
+                        return
+                    case 'CREATE_FAILED':
+                    case 'PUBLISH_FAILED':
+                        // Failed - reject with error
+                        return reject(new Error('Deployment failed.'))
+                    default:
+                        // Not in progress or failed - assume success
```

Comment:
Why do we assume success if "not in progress or failed"? Sounds backwards

Reply:
🤷 I just copied the logic from the [existing code](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1429/files#diff-d1c984792e22d7551aa2dbca33b9cb0014bf6bad55e4faa8474e391a7f188231). But looking at the back end code, it seems like the only other state a target can be is `ACTIVE`.